### PR TITLE
Don't lose materialization interval tracking when re-applying feature views

### PR DESCRIPTION
**Pull request by jklegar**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: sdk/python/feast/registry.py [hunk 1]
```diff
@@ -185,11 +185,60 @@ def updater(registry_proto: RegistryProto):
                     == feature_view_proto.spec.name
                     and existing_feature_view_proto.spec.project == project
                 ):
+                    # do not update if feature view has not changed; updating will erase tracked materialization intervals
```
Comment (woop) [inline sdk/python/feast/registry.py:188]: It's not immediately clear to me what is happening here. Mind explaining?

Comment (jklegar) [reply sdk/python/feast/registry.py:188]: Yeah, this function `apply_feature_view` does a "destructive update" - it erases tracked materialization intervals so as far as Feast is aware, no time interval of this feature view has been materialized yet. This is what we want if we changed the feature view, but if we didn't it'll cause unnecessary rematerializations. So this part just says don't do a destructive update if the feature view is exactly the same as what's already in the registry

**Review by achals — COMMENTED** [review]
nits, but looks good otherwise.

**Review by achals — APPROVED** [review]
/lgtm

**Review by achals — APPROVED** [review]
/lgtm

**Pull request merged**